src/Symfony/Component/HttpClient/Response/CurlResponse.php

```diff
@@ -320,7 +320,20 @@ private static function perform(ClientState $multi, ?array &$responses = null):
                 }
 
                 $multi->handlesActivity[$id][] = null;
-                $multi->handlesActivity[$id][] = \in_array($result, [\CURLE_OK, \CURLE_TOO_MANY_REDIRECTS], true) || '_0' === $waitFor || curl_getinfo($ch, \CURLINFO_SIZE_DOWNLOAD) === curl_getinfo($ch, \CURLINFO_CONTENT_LENGTH_DOWNLOAD) || (curl_error($ch) === 'OpenSSL SSL_read: SSL_ERROR_SYSCALL, errno 0' && -1.0 === curl_getinfo($ch, \CURLINFO_CONTENT_LENGTH_DOWNLOAD) && \in_array('close', array_map('strtolower', $responses[$id]->headers['connection']), true)) ? null : new TransportException(ucfirst(curl_error($ch) ?: curl_strerror($result)).sprintf(' for "%s".', curl_getinfo($ch, \CURLINFO_EFFECTIVE_URL)));
+                $multi->handlesActivity[$id][] =
+                    (
+                        \in_array($result, [\CURLE_OK, \CURLE_TOO_MANY_REDIRECTS], true)
+                        || '_0' === $waitFor
+                        || curl_getinfo($ch, \CURLINFO_SIZE_DOWNLOAD) === curl_getinfo($ch, \CURLINFO_CONTENT_LENGTH_DOWNLOAD)
+                        || (
+                            'OpenSSL SSL_read: SSL_ERROR_SYSCALL, errno 0' === curl_error($ch)
+                            && -1.0 === curl_getinfo($ch, \CURLINFO_CONTENT_LENGTH_DOWNLOAD)
+                            && \array_key_exists('connection', $responses[$id]->headers)
+                            && \in_array('close', array_map('strtolower', $responses[$id]->headers['connection']), true)
```

Comment:
so, the question to me is: do we need these lines at all? maybe ignoring the SSL error is all we should do ("errno 0" is a hint this is not an issue I guess)
ideally, one of you would give it a try and report back the results from live usage.
Up to it?